### PR TITLE
Bad detections fix

### PR DESCRIPTION
**Pull request by LiranBratt2121**
- Add false positives removal function
- Add lost robot ids finder function (only works when there is only one lost robot in a single frame)
- Add function documentation to: `sd_heatmap_utils`, `sd_tracking_utils`, `sd_utils` 
- Add Nvidia-Cuda support
- Update dependeces pytorch, torchvision, torchaudio and ultralytics to the following versions:  
   * supervision==0.15.0rc4
   * torch==2.1.0+cu121
   * torchaudio==2.1.0+cu121
   * torchvision==0.16.0+cu121
   * ultralytics==8.0.132

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,86 +1,113 @@
...
-                
-            lost_robots = set(id_to_robot_number.values()) - set(current_robots)
-            current_robots.clear()
-                
+                x1, y1, x2, y2 = detection
+
+                try:
+                    center_x = (x1 + x2) // 2
+                    center_y = (y1 + y2) // 2
+
```
Comment (NoamRosenberg08) [inline main.py:82]: move to function to the center of a robot

File: sd_tracking_utils.py [hunk 2]
```diff
@@ -47,3 +53,225 @@ def map_tracker_ids(robot_mapping, id_list):
...
+        ----
+            - data (list): A list of data tensors to be updated.
+            - result (Results): The YOLO Results object containing detection data.
+
+        Returns:
+        ----
+            - list: The updated list of data tensors.
+
+        """
+        for idx in range(6):
```
Comment (NoamRosenberg08) [inline sd_tracking_utils.py:217]: what is 6? write a Constant for the number of robots, what will happend if we will go back to the corn game?

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (sagi21805) [reply main.py:82]: I don't know if Liran added such functions, but in my branch I think in utils there is a center function.

Comment (LiranBratt2121) [reply main.py:82]: So should we merge our branches into dev and make this fix there?

**Pull request merged**